src/Auth/CanDirective.php

```diff
@@ -189,7 +193,7 @@ protected function modelsToCheck(ArgumentSet $argumentSet, array $args): iterabl
                 $queryBuilder->withTrashed();
             }
 
-            $directivesContainsRestore = $argumentSet->directives->contains(
+            $directivesContainsRestore = $resolveInfo->argumentSet->directives->contains(
```

Comment:
`$resolveInfo->argumentSet->directives` is repeated

Reply:
Made it them both into one variable with the name of `$argumentSetDirectives` if you have a better name in mind please let me know.